apps/barista-design-system/src/components/content-link/content-link.ts

```diff
@@ -0,0 +1,93 @@
+/**
+ * @license
+ * Copyright 2020 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  Component,
+  Input,
+  HostListener,
+  HostBinding,
+  OnChanges,
+  OnInit,
+} from '@angular/core';
+import { Router, Event, NavigationEnd } from '@angular/router';
+
+/**
+ * The ba-content link component is used because we need to dynamically
+ * instanciate a router link and this is not possible, because it is a directive.
+ * So we create a component that handles the navigation.
+ */
+@Component({
+  selector: 'a[contentLink]',
+  template: '<ng-content></ng-content>',
+})
+//TODO: Implement HTML sanitizer in on init for linkValue if <ng-content> doesn't work.
+export class BaContentLink implements OnChanges, OnInit {
+  /** Absolute url for navigation on the page. For example /components/button */
+  @Input() contentLink: string;
+
+  /** Href link to display */
+  @HostBinding() href: string;
+
+  /** QueryParams of absolute url */
+  @Input() queryParams: { [k: string]: string };
+
+  /** Fragement of absolute url */
+  @Input() fragment: string;
+
+  constructor(private _router: Router) {
+    //todo unsubscribe
+    this._router.events.subscribe((navigation: Event) => {
+      if (navigation instanceof NavigationEnd) {
+        this._updateTargetUrlAndHref();
+      }
+    });
+  }
+
+  ngOnInit(): void {
+    this._updateTargetUrlAndHref();
+  }
+
+  ngOnChanges(): void {
+    this._updateTargetUrlAndHref();
+  }
+
+  @HostListener('click', [
+    '$event.button',
+    '$event.ctrlKey',
+    '$event.metaKey',
+    '$event.shiftKey',
+  ])
```

Comment:
I couldn't find a way to get the arguments ($event.button, etc.) when using the decorator.
Is there a way to do that?

Reply:
@tomheller there is only one argument why you should define it in the decorator. This is when you have static property, that does not change. Then you would have to create a variable to bind it with a `@HostBinding` only to accomplish that. For that, the host property in the decorator is cleaner because you don't have to add additional logic to your component.
We said for the library it makes sense to have this convention because every component has a class on the host that is static. 

For apps I don't see any benefit - especially in this case I would prefer the decorator as it sticks to the class and makes the code more readable. Furthermore, it is originally the recommended way from angular to accomplish that.

Maybe we can take an action item out of that a discussion in the CODING_STANDARDS that mark this.

Reply:
@lukasholzer the onClick is a static function that is not changed. You can just as easily do that in the component decorator and pass the event in, there are no additional bindings needed. 

It is in the CODING_STANDARDS within the repository, this makes it apply to everything that is in the repository. 
If you want to discuss these things you can open a discussion issue. 

It would be the first introduction of a `@HostBinding` in the whole codebase and I am arguing that it could be done within the set guidelines. But for this pull request, I don't care...

Reply:
I agree with @tomheller currently it is in our coding standards and we should stick to it until we have a discussion about it.